### PR TITLE
Develop register verification view.

### LINKED ISSUE #40: Register Verification View
### Description 
Create a view where users verify their registration through a verification code or link.

Tasks

- Design the UI for registration verification.
- Integrate backend logic to handle registration verification.
- Manage cases like expired verification links or incorrect codes.
- Allow users to request a new verification link if needed.

### Design
- image

### PR DESCRIPTION
**Pull request by asmaaadel0**
- Description:
  - Create a view where users verify their registration through a verification code or link.

- Issue:
  - https://github.com/codescalersinternships/Flyspray/issues/40

- Design:
  - if user submit form without entering the 4 digits of the code
image

  - if user submit form successfully
image

### CODE REVIEW

File: frontend/src/components/auth/VerificationComponent.vue [hunk 1]
```diff
@@ -0,0 +1,176 @@
+<template>
+  <div class="login-box">
+    <div class="header-box">
+      <p class="header-text" id="login-card-title">Verify Your Email</p>
+      <div class="sub-box">
+        <p class="sub-header-text" id="verification-card-title">
+          Please enter the 4 digit code sent to your email your...@domain.com.
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/VerificationComponent.vue]:
```suggestion
          Please enter the 4-digit code sent to your email.
```

File: frontend/src/components/auth/VerificationComponent.vue [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        if (this.code[index].length > 1) {
+          this.code[index] = this.code[index].slice(0, 1);
+        }
+      }
+
+      //   for (let index = 0; index < 3; index++) {
+      //     if (this.code[index].length === 1) {
+      //       (this.$refs["digit" + (index + 1)] as any)[0].focus();
+      //     }
+      //   }
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/VerificationComponent.vue]: Please remove the commented code if not needed

File: frontend/src/components/auth/VerificationComponent.vue [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+import { defineComponent } from "vue";
+import InputForm from "./InputForm.vue";
+
+export default defineComponent({
+  components: {
+    InputForm,
+  },
+  data() {
+    return {
+      code: ["", "", "", ""] as string[],
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/VerificationComponent.vue]: Why string? its 4 digits

File: frontend/src/components/auth/VerificationComponent.vue [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      error: false as boolean,
+    };
+  },
+  methods: {
+    formatcode() {
+      for (let index = 0; index < 4; index++) {
+        if (this.code[index].length > 1) {
+          this.code[index] = this.code[index].slice(0, 1);
+        }
+      }
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/VerificationComponent.vue]: Also here, we should accept 4 digits then make it required instead

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]
Good job, I've added some comments please take a look at them.

**Review by Mahmoud-Emad — APPROVED** [review]

**Pull request merged**